networking/configuring-ipfailover.adoc

```diff
@@ -0,0 +1,62 @@
+
+[NOTE]
+====
+The VIPs must be routable from outside the cluster.
+====
+
+IP failover monitors a port on each VIP to determine whether the port is reachable on the node. If the port is not reachable, the VIP is not assigned to the node. If the port is set to `0`, this check is suppressed. The check script does the needed testing.
+
+IP failover uses link:http://www.keepalived.org/[Keepalived] to host a set of externally accessible VIP addresses on a set of hosts. Each VIP is only serviced by a single host at a time. Keepalived uses the Virtual Router Redundancy Protocol (VRRP) protocol to determine which host, from the set of hosts, services which VIP. If a host becomes unavailable or if the service that Keepalived is watching does not respond, the VIP is switched to another host from the set, which means a VIP is always serviced as long as a host is available.
+
+When a host running Keepalived passes the check script, the host can become in the master state based on its priority and the priority of the current master, as determined by the preemption strategy.
+
+A cluster administrator can provide a script though the `OPENSHIFT_HA_NOTIFY_SCRIPT` variable, which is called whenever the state changes. Keepalived is in master state when it is servicing the VIP, in backup state when another node is servicing the VIP, or in fault state when the check script fails. The notify script is called with the new state whenever the state changes.
+
+You can create an IP failover deployment configuration on {product-title}. The IP failover deployment configuration specifies the set of VIP addresses, and the set of nodes on which to service them. A cluster can have multiple IP failover deployment configurations, with each managing its own set of unique VIP addresses. Each node in the IP failover configuration runs an IP failover pod, and this pod runs Keepalived.
+
+When using VIPs to access a pod with host networking, the application pod runs on all nodes that are running the IP failover pods. This enables any of the IP failover nodes to become the master and service the VIPs when needed. If application pods are not running on all nodes with IP failover, either some IP failover nodes never service the VIPs or some application pods never receive any traffic. Use the same selector and replication count, for both IP failover and the application pods, to avoid this mismatch.
+
+While using VIPs to access a service, any of the nodes can be in the IP failover set of nodes, since the service is reachable on all nodes, no matter where the application pod is running. Any of the IP failover nodes can become master at any time. The service can either use external IPs and a service port or it can use a `NodePort`.
```

Comment:
Here, we're talking about a Kubernetes "Service", right?  We should capitalize per the convention to make it clear that we are talking about a service in the Kubernetes sense:
```suggestion
While using VIPs to access a Service, any of the nodes can be in the IP failover set of nodes, since the Service is reachable on all nodes, no matter where the application Pod is running. Any of the IP failover nodes can become master at any time. The Service can either use external IPs and a Service port or it can use a `NodePort`.
```